# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasRenderer.h
# Commit: 7fea32057b92
# Full Hash: 7fea32057b920eac49960541efa2f3fea9074c19
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754130
# File Overlap Count: 2
# Description:
#   Bug 1754130 - Support presenting a WebGLFramebuffer to its own swap chain without opaque FB. r=aosmond,jgilbert
#   
#   Most of the support for presenting a WebGLFramebuffer to a swap chain existed as part of the
#   mechanism for opaque WebXR framebuffer support. However, such "opaque" framebuffer are meant
#   to be opaque in the sense that their attachments can't be inspected or changed, which does
# ==============================================================================

diff -r 1ac8f10e0025 -r 7fea32057b92 gfx/layers/CanvasRenderer.h
--- a/gfx/layers/CanvasRenderer.h	Fri Feb 11 15:36:30 2022 +0000
+++ b/gfx/layers/CanvasRenderer.h	Fri Feb 11 15:36:30 2022 +0000
@@ -134,6 +134,9 @@
   std::shared_ptr<BorrowedSourceSurface> BorrowSnapshot(
       bool requireAlphaPremult = true) const;
 
+  virtual bool CopySnapshotTo(gfx::DrawTarget* aDT,
+                              bool aRequireAlphaPremult = true);
+
   void FirePreTransactionCallback() const;
   void FireDidTransactionCallback() const;
 };